File: 2021-SOPM/E1/E104UIGreeting/app/src/main/java/com/example/e104_ui_greeting/MainActivity.java
package com.example.e104_ui_greeting;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import com.google.android.material.button.MaterialButton;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        MaterialButton loginButton = findViewById(R.id.login_button);

        loginButton.setOnClickListener(view -> Toast.makeText(MainActivity.this, "test", Toast.LENGTH_LONG).show());

    }
}

File: 2022-SOPM/E04/app/src/main/java/com/example/e04_00/MainActivity.kt
package com.example.e04_00

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.Toast
import com.example.e04_00.utilities.PASSWORD_KEY
import com.example.e04_00.utilities.USERNAME_KEY

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val button: Button = findViewById(R.id.login_button)
        button.setOnClickListener {
            handleOnButtonClick()
        }
    }

    private fun handleOnButtonClick() {
        val usernameEditText: EditText = findViewById(R.id.username)
        val passwordEditText: EditText = findViewById(R.id.password)

        val username = usernameEditText.text.toString()
        val password = passwordEditText.text.toString()

        Toast.makeText(this, "Username: $username, Password: $password", Toast.LENGTH_LONG).show()

        val intent = Intent(this, ContentActivity::class.java)
        intent.putExtra(USERNAME_KEY, username)
        intent.putExtra(PASSWORD_KEY, password)
        startActivity(intent)
        finish()
    }
}